internal/venice-common/src/main/java/com/linkedin/venice/system/store/ControllerClientBackedSystemSchemaInitializer.java

```diff
@@ -236,11 +251,20 @@ private void checkAndMayRegisterValueSchema(
       String storeName,
       int valueSchemaId,
       Schema schemaInZk,
-      Schema schemaInLocalResources) {
+      Schema schemaInLocalResources,
+      boolean skipCompatibilityCheck) {
     if (schemaInZk == null) {
-      SchemaResponse addValueSchemaResponse = controllerClient.retryableRequest(
-          DEFAULT_RETRY_TIMES,
-          c -> c.addValueSchema(storeName, schemaInLocalResources.toString(), valueSchemaId));
+      SchemaResponse addValueSchemaResponse = controllerClient.retryableRequest(DEFAULT_RETRY_TIMES, c -> {
+        if (skipCompatibilityCheck) {
+          return c.addValueSchema(
+              storeName,
+              schemaInLocalResources.toString(),
+              valueSchemaId,
+              DirectionalSchemaCompatibilityType.NONE);
+        }
+        return c.addValueSchema(storeName, schemaInLocalResources.toString(), valueSchemaId);
+      });
```

Comment:
KME schemas are backward compatible. We'd better enforce backward compatibility `DirectionalSchemaCompatibilityType.BACKWARD` when registering a new KME. Note it varies for different schemas. For example, meta system store schema needs to be registered with ` DirectionalSchemaCompatibilityType.FULL`. You may need to add ` DirectionalSchemaCompatibilityType` as a parameter of the function or a member of the class.

Reply:
Sounds good. I will fix this.